graphql/src/store/resolver.rs

```diff
@@ -288,21 +287,7 @@ where
                     })?,
                     _ => None,
                 },
-                _ => {
-                    let mut query = build_query(&object_type, arguments)?;
-
-                    // Add matching filter for derived fields
-                    Self::add_filter_for_derived_field(
-                        &mut query,
-                        parent,
-                        field_definition,
-                        object_type,
-                    );
-
-                    query.range = Some(EntityRange { first: 1, skip: 0 });
-
-                    self.store.find(query)?.into_iter().next()
-                }
+                _ => panic!("top level queries must either take an `id` or return a list"),
```

Comment:
If this can't be hit by a user query then I'm ok with the `panic!`, otherwise I'd prefer returning a `QueryExecutionError`.

Reply:
It's impossible for a user to hit this because the `Query` type is given by us, and all generated fields in it either take an `id` argument for querying a single object or return a list when querying multiple objects. Does that sound right to you?

Reply:
Yep!